src/types.ts

```diff
@@ -122,28 +122,33 @@ export type LimitedFields<T, F extends keyof T> = { [P in F]: T[P] };
 export const isObject = (o: {}): boolean =>
   o instanceof Object && o.constructor === Object;
 
+/**
+ * Return an object mantaining keys that satisfy the given processor function.
+ */
+export const filterValues = (f: (t: RecursiveRecord<unknown>) => boolean) => <
+  T
+>(
+  obj: T
+): T =>
```

Comment:
This type is wrong, the returned value is not `T`, but a subset dependent from the `f` function. The problem will be on nested object too.